#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From e31b6670e3351fc1955cdc22ef9a8f5decb69a5c Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Tue, 2 Jul 2024 14:45:09 +0200
Subject: [PATCH] Make the depedency to QML optional

Add the option to disable a dependency to Qt QML
and explicitly check for it if QML is used.
---
 CMakeLists.txt     | 5 +++++
 src/CMakeLists.txt | 4 +++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd23be5e..3e962e51 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,8 @@ set(EXCLUDE_DEPRECATED_BEFORE_AND_AT 0 CACHE STRING "Control the range of deprec
 option(BUILD_QCH "Build API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)" OFF)
 add_feature_info(QCH ${BUILD_QCH} "API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)")

+option(KCONTACTS_USE_QML "Build the QML plugin" ON)
+
 set(KCONTACTS_LIB_VERSION ${KF_VERSION})

 set(kcontacts_version_header "${CMAKE_CURRENT_BINARY_DIR}/src/kcontacts_version.h")
@@ -51,6 +53,9 @@ set_package_properties(Qt6Gui PROPERTIES
     TYPE REQUIRED
     PURPOSE "Basic library component"
 )
+if (KCONTACTS_USE_QML)
+    find_package(Qt6 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Qml)
+endif()
 find_package(KF6 ${KF_DEP_VERSION} REQUIRED COMPONENTS CoreAddons I18n Config Codecs)

 ########### Targets ###########
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bc72fc6f..43e23674 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -168,7 +168,9 @@ ecm_generate_headers(KContacts_CamelCase_HEADERS
   REQUIRED_HEADERS KContacts_HEADERS
 )

-add_subdirectory(qml)
+if (KCONTACTS_USE_QML)
+  add_subdirectory(qml)
+endif ()
 add_subdirectory(converter)

 install(FILES
--
2.45.2
